ci-operator/config/openstack-k8s-operators/ci-framework/openstack-k8s-operators-ci-framework-master.yaml

```diff
@@ -0,0 +1,21 @@
+build_root:
```

Comment:
Currently, You are using golang-1.13 container as a base container.
~~~
build_root:
  image_stream_tag:
    name: release
    namespace: openshift
    tag: golang-1.13 >>>
~~~

~~~
- as: tests
  commands: make tests
  container:
    from: src  >>>
~~~    



We need to use an image that suits our test requirement, We have a few options.

* Create an image using dockerfile_literal/file:

ex: https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/nova-operator/openstack-k8s-operators-nova-operator-master.yaml#L12-L19

* Reuse an image which already has the deps we need. 

fyi.. We do create an image for our e2e job/precommit job via openstack-k8s-operators-ci[1] if you want to reuse it.

To use it add a .ci-operator.yaml in ci-framerwork repo, example[2]

then you need to add `from_repository: true` and use src as base
~~~
build_root:
  from_repository: true
.
.  
- as: tests
  commands: make tests
  container:
    from: src 
~~~  

If you have to add more deps on src(base container), then you can follow this kind of pattern to add additional deps[3] and use that container[4]


[1] https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/blob/master/images/build-root/Dockerfile.custom
[2] https://github.com/openstack-k8s-operators/openstack-operator/blob/master/.ci-operator.yaml
[3] https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/openstack-ansibleee-operator/openstack-k8s-operators-openstack-ansibleee-operator-main.yaml#L14-L21
[4] https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/openstack-ansibleee-operator/openstack-k8s-operators-openstack-ansibleee-operator-main.yaml#L83-L84

Reply:
Thanks for the help @sandeepyadav93 !